public/pages/Rules/components/RuleEditor/DetectionVisualEditor.tsx

```diff
@@ -346,15 +351,27 @@ export class DetectionVisualEditor extends React.Component<
         });
       }
     }
-    errors.touched['condition'] = true;
 
-    const detectionObj = { ...this.state.detectionObj, condition: value } as DetectionObject;
+    errors.touched['condition'] = true;
     this.setState({
-      detectionObj,
       errors,
     });
   };
 
+  private updateCondition = (value: string) => {
+    value = value.trim();
+
+    const detectionObj = { ...this.state.detectionObj, condition: value } as DetectionObject;
```

Comment:
why do we need this cast? Can we instead do `const detectionObj: DetectionObj = { ... }`

Reply:
Sure, will be refactored.